### PR TITLE
Add log out context menu (#44)

### PR DESCRIPTION
**Pull request by shmato4ek**

### CODE REVIEW

File: frontend/src/app/modules/workspace/navbar/navbar.component.ts [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+    this.contextMenuPosition.x = window.outerWidth - 50 + 'px';
+    this.contextMenuPosition.y = 100 + 'px';
+    this.contextMenu?.menu.focusFirstItem('mouse');
+    this.contextMenu?.openMenu();
+  }
 
+  onLogOut() {
+    localStorage.removeItem('accessToken');
+    localStorage.removeItem('refreshToken');
+  }
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/workspace/navbar/navbar.component.ts]: Move it to AuthService, add redirection to /login

File: frontend/src/app/modules/workspace/navbar/navbar.component.ts [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
+    event.preventDefault();
+    this.contextMenuPosition.x = window.outerWidth - 50 + 'px';
+    this.contextMenuPosition.y = 100 + 'px';
+    this.contextMenu?.menu.focusFirstItem('mouse');
+    this.contextMenu?.openMenu();
+  }
 
+  onLogOut() {
+    this.loginService.logOut();
+    this.router.navigate([`/login`]);
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/workspace/navbar/navbar.component.ts]: Also to loginService

File: frontend/src/app/modules/workspace/navbar/navbar.component.ts [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
+  }
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/workspace/navbar/navbar.component.ts:22]: Github warning

File: frontend/src/app/modules/workspace/navbar/navbar.component.ts [hunk 4]
```diff
@@ -1,12 +1,29 @@
...
+  @ViewChild(MatMenuTrigger)
+  contextMenu?: MatMenuTrigger;
 
-  constructor() { }
+  contextMenuPosition = { x: '0px', y: '0px' };
+  constructor(private loginService: LoginService, private router: Router) {}
+  onContextMenu(event: MouseEvent) {
+    event.preventDefault();
+    this.contextMenuPosition.x = window.outerWidth - 50 + 'px';
+    this.contextMenuPosition.y = 100 + 'px';
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/workspace/navbar/navbar.component.ts]: Can't it be done via a material context menu?

Comment (Jyuart) [reply frontend/src/app/modules/workspace/navbar/navbar.component.ts]: Yes, make it easier. And it should handle left-button click

**Review by Anatolii-Ivanov — DISMISSED** [review]

File: frontend/src/app/modules/workspace/navbar/navbar.component.html [hunk 5]
```diff
@@ -1,15 +1,38 @@
...
+        style="visibility: hidden; position: fixed"
+        [style.left]="contextMenuPosition.x"
+        [style.top]="contextMenuPosition.y"
+        [matMenuTriggerFor]="contextMenu"
+      ></div>
+      <mat-menu #contextMenu="matMenu">
+        <ng-template matMenuContent>
+          <button mat-menu-item (click)="onLogOut()">Log out</button>
+        </ng-template>
+      </mat-menu> -->
```
Comment (Jyuart) [inline frontend/src/app/modules/workspace/navbar/navbar.component.html]: Please do not leave a commented code in you PR

**Review by Jyuart — DISMISSED** [review]

**Review by Jyuart — APPROVED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Pull request merged**